src/main/java/hr/fer/dsd/privtap/service/TriggerListener.java

```diff
@@ -24,13 +26,24 @@ public void onApplicationEvent(TriggerEvent event){
                 .stream()
                 .filter(a -> a
                         .getTrigger()
-                        .getId()
+                        .getTypeId()
                         .equals(event.getTrigger().getTypeId()))
                 .collect(Collectors.toSet());
         for(var a : automations){
-            var action = actionService.getByTypeAndUser(a.getAction().getId(),userId);
-            //action.setFields();
-            //set actionFields and call a method to send the action
+            var action = a.getAction();
+            System.out.println(action.getName());
+            var fields = new ArrayList<RequestField>();
+            for(RequestField f : action.getFields()){
+                var eachField = ((RequestField)f.getName().getRelatedClass()).buildDefault(f.getName());
+                for(RequestField triggerField : event.getTrigger().getFields()){
```

Comment:
Maybe use stream on event.getTrigger().getFields() to avoid using three for loops, it could improve readability.